### ISSUE TITLE
After install `zsh` from Homebrew, `chsh` fails

### ISSUE THREAD

**Issue by sharplet** [thoughtbot/laptop#447]
1. Run script for first time, with login shell set to `bash`
2. Script bails out at some point after install `zsh` from Homebrew
3. Re-run the script. At this point `SHELL` hasn't changed to `zsh` because we're in the same shell session as before.
4. `chsh -s "$(which zsh)"` fails with the error `chsh: /usr/local/bin/zsh: non-standard shell`

Relevant SO post: http://superuser.com/questions/362372/how-to-change-the-login-shell-on-mac-os-x-from-bash-to-zsh

The answers there suggest one of two approaches:
- Use `dscl` instead of `chsh`, which is equivalent to setting the shell via the Accounts preference pane, and allows a non-standard shell to be set.
- Ensure `/usr/local/bin/zsh` is added to `/etc/shells`.

I managed to get something working with `dscl` (unfortunately it's more complicated than `chsh` because if `UserShell` has never been set, `dscl . change` fails instead of creating it for you):

```
create_or_change_login_shell() {
  local new_shell="$1"
  local current_shell="$(dscl . read "$HOME" UserShell 2>/dev/null | cut -d" " -f2)"

  if [ -z "$current_shell" ]; then
    sudo dscl . create "$HOME" UserShell "$new_shell"
  elif [ "$current_shell" = "$new_shell" ]; then
    return
  else
    sudo dscl . change "$HOME" UserShell "$current_shell" "$new_shell"
  fi
}
```

However it's probably much simpler to just append `/usr/local/bin/zsh` to `/etc/shells`.

**Comment by sharplet** [comment]
#440 already takes the approach of appending to `/etc/shells`.

**Comment by awolfson** [comment]
I had the same issue. In case anyone needs the easy fix. All I had to do was log out of my shell and log back in. When I logged back in, it was running `zsh`.

**Comment by croaky** [comment]
Think we fixed this in #440.

Also currently considering `sudo chsh -s "$shell_path" "$USER"` in https://github.com/thoughtbot/laptop/pull/501
